src/ripple/app/paths/impl/BookStep.cpp

```diff
@@ -533,10 +533,10 @@ BookStep<TIn, TOut, TDerived>::forEachOffer (
                 // Issuer requires authorization.  See if offer owner has that.
                 auto const& ownerID = offer.owner();
                 auto const authFlag =
-                    ownerID > issuerID ? lsfHighAuth : lsfLowAuth;
+                    ownerID > issuerID ? lsfLowAuth : lsfHighAuth;
 
                 auto const line = afView.read (keylet::line (
-                    ownerID, issuerID, offer.issueOut().currency));
+                    ownerID, issuerID, offer.issueIn().currency));
```

Comment:
Ah, your right, it does delete it on a missing trust line. I actually think we do want a check for XRP at the top of this, even though the code is correct as-is. It makes it clearer.